### PR TITLE
LibWeb + LibJS: Fix WindowObject functions and other things for css3test

### LINKED ISSUE #8989: Browser: crashes when opening google.com
On commit 2002b7e2e366975e0fa54a86dc42a51fae1e15d0

Log:
```
WebContent(35:35): FrameLoader::load: https://www.google.com/
WebContent(35:35): Favicon downloaded, 5430 bytes from https://www.google.com/favicon.ico
WebContent(35:35): Decoded favicon, [32x32]
WebContent(35:35): This content has MIME type 'text/html', encoding 'iso-8859-1'
WebContent(35:35): CSSParser: Unrecognized property '_background-image'
WebContent(35:35): CSSParser: Unrecognized property '_height'
WebContent(35:35): CSSParser: Unrecognized property 'filter'
WebContent(35:35): CSSParser: Unrecognized property 'filter'
WebContent(35:35): CSSParser: Unrecognized property 'box-shadow'
WebContent(35:35): CSSParser: Unrecognized property 'filter'
WebContent(35:35): CSSParser: Unrecognized property '*opacity'
WebContent(35:35): CSSParser: Unrecognized property '*top'
WebContent(35:35): CSSParser: Unrecognized property '*left'
WebContent(35:35): CSSParser: Unrecognized property '*right'
WebContent(35:35): CSSParser: Unrecognized property '*bottom'
WebContent(35:35): CSSParser: Unrecognized property 'zoom'
WebContent(35:35): CSSParser: Unrecognized property 'zoom'
WebContent(35:35): CSSParser: Unrecognized property '*display'
WebContent(35:35): CSSParser: Unrecognized property 'box-shadow'
WebContent(35:35): CSSParser: Unrecognized property '*display'
WebContent(35:35): CSSParser: Unrecognized property '*display'
WebContent(35:35): CSSParser: Unrecognized property 'zoom'
WebContent(35:35): CSSParser: Unrecognized property '_background'
WebContent(35:35): CSSParser: Unrecognized property '*padding'
WebContent(35:35): CSSParser: Unrecognized property '_background-image'
WebContent(35:35): CSSParser: Unrecognized property 'outline'
WebContent(35:35): CSSParser: Unrecognized property '*padding'
WebContent(35:35): CSSParser: Unrecognized property '*padding'
WebContent(35:35): CSSParser: Unrecognized property '*display'
WebContent(35:35): CSSParser: Unrecognized property '_background'
WebContent(35:35): CSSParser: Unrecognized property '*display'
WebContent(35:35): CSSParser: Unrecognized property 'outline'
WebContent(35:35): CSSParser: Unrecognized property 'outline'
WebContent(35:35): CSSParser: Unrecognized property 'outline'
WebContent(35:35): Unknown pseudo class: 'vertical'
WebContent(35:35): CSSParser: Unrecognized property 'box-shadow'
WebContent(35:35): CSSParser: Unrecognized property 'content'
WebContent(35:35): CSSParser: Unrecognized property '*zoom'
WebContent(35:35): CSSParser: Unrecognized property '*display'
WebContent(35:35): CSSParser: Unrecognized property 'filter'
WebContent(35:35): CSSParser: Unrecognized property '*border-collapse'
WebContent(35:35): CSSParser: Unrecognized property '*white-space'
WebContent(35:35): CSSParser: Unrecognized property '*border-collapse'
WebContent(35:35): CSSParser: Unrecognized property '*display'
WebContent(35:35): CSSParser: Unrecognized property '*min-width'
WebContent(35:35): CSSParser: Unrecognized property 'box-shadow'
WebContent(35:35): CSSParser: Unrecognized property 'outline'
WebContent(35:35): CSSParser: Unrecognized property 'box-shadow'
WebContent(35:35): CSSParser: Unrecognized property 'box-shadow'
WebContent(35:35): CSSParser: Unrecognized property 'box-shadow'
WebContent(35:35): CSSParser: Unrecognized property 'filter'
WebContent(35:35): CSSParser: Unrecognized property 'box-shadow'
WebContent(35:35): CSSParser: Unrecognized property 'filter'
WebContent(35:35): CSSParser: Unrecognized property 'filter'
WebContent(35:35): CSSParser: Unrecognized property 'filter'
WebContent(35:35): CSSParser: Unrecognized property 'filter'
WebContent(35:35): CSSParser: Unrecognized property 'box-shadow'
WebContent(35:35): CSSParser: Unrecognized property 'box-shadow'
WebContent(35:35): CSSParser: Unrecognized property '*zoom'
WebContent(35:35): CSSParser: Unrecognized property 'content'
WebContent(35:35): CSSParser: Unrecognized property 'outline'
WebContent(35:35): CSSParser: Unrecognized property 'outline'
WebContent(35:35):   CSS::Selector:
    {None} TagName:li
    {ImmediateChild} TagName:a, Class:gbmt

WebContent(35:35): ASSERTION FAILED: false
./Userland/Libraries/LibWeb/Bindings/WindowObject.cpp:107
[#0 WebContent(35:35)]: Terminating WebContent(35) due to signal 6
[#0 FinalizerTask(4:4)]: 0xdeadc0de  Kernel::Processor::switch_context(Kernel::Thread*&, Kernel::Thread*&) + 0x6ac
0xdeadc0de  Kernel::Scheduler::context_switch(Kernel::Thread*) [clone .localalias] + 0x40e
0xdeadc0de  Kernel::Scheduler::pick_next() [clone .localalias] + 0x2c2
0xdeadc0de  Kernel::Processor::check_invoke_scheduler() [clone .localalias] + 0x219
0xdeadc0de  Kernel::Thread::die_if_needed() [clone .localalias] + 0x232
0xdeadc0de  syscall_handler + 0x275f
0xdeadc0de  syscall_asm_entry + 0x31
0x8eff7135  libc.so: .text + 0x2a135
0x8efe0f46  libc.so: .text + 0x13f46
0x8efe1a0a  libc.so: .text + 0x14a0a
0xa7f6c89e  libweb.so: .text + 0x12a89e
0xa7f6f25e  libweb.so: .text + 0x12d25e
0x0fa5c12f  libjs.so: .text + 0x10712f
0x0fadc2fa  libjs.so: .text + 0x1872fa
0x0fbb2722  libjs.so: .text + 0x25d722
0x0f9d78b4  libjs.so: .text + 0x828b4
0x0f9ce1ff  libjs.so: .text + 0x791ff
0x0fa114b0  libjs.so: .text + 0xbc4b0
0x0faf2e5e  libjs.so: .text + 0x19de5e
0x0faf382a  libjs.so: .text + 0x19e82a
0x0fbb2722  libjs.so: .text + 0x25d722
0x0f9d78b4  libjs.so: .text + 0x828b4
0x0f9ce1ff  libjs.so: .text + 0x791ff
0x0fa114b0  libjs.so: .text + 0xbc4b0
0x0f9cdf0c  libjs.so: .text + 0x78f0c
0x0fa11a36  libjs.so: .text + 0xbca36
0xa7fe8460  libweb.so: .text + 0x1a6460
0xa8051a1b  libweb.so: .text + 0x20fa1b
0xa8052b55  libweb.so: .text + 0x210b55
0xa806f56d  libweb.so: .text + 0x22d56d
0xa8080a4c  libweb.so: .text + 0x23ea4c
0xa80e0683  libweb.so: .text + 0x29e683
0xa80e3a42  libweb.so: .text + 0x2a1a42
0xa80eaf71  libweb.so: .text + 0x2a8f71
0xa80ec9a3  libweb.so: .text + 0x2aa9a3
0xa80ee94f  libweb.so: .text + 0x2ac94f
0xa80f1481  libweb.so: .text + 0x2af481
0x1aa1a209  libprotocol.so: .text + 0x15209
0x1aa17cc2  libprotocol.so: .text + 0x12cc2
0x8358e78a  libcore.so: .text + 0x3f78a
0x8358fd91  libcore.so: .text + 0x40d91
0x8357735a  libcore.so: .text + 0x2835a
0x83577b6b  libcore.so: .text + 0x28b6b
0x11d64474  /bin/WebContent: .text + 0x12474
0x11d645c8  /bin/WebContent: .text + 0x125c8

[#0 FinalizerTask(4:4)]: Generating coredump for pid: 35
[FinalizerTask(4:4)]: RTC: Year: 2021, month: 7, day: 24, hour: 20, minute: 20, second: 3
CrashDaemon(21:21): New coredump file: /tmp/coredump/WebContent_35_1627158003
[#0 NetworkTask(7:7)]: flush_delayed_tcp_acks: 1 sockets remaining
[#0 NetworkTask(7:7)]: handle_tcp: unexpected flags in FinWait2 state (10)
WebContent(38:38): Parse error! [handle_initial @ ./Userland/Libraries/LibWeb/HTML/Parser/HTMLDocumentParser.cpp:336]
CrashDaemon(21:21): --- Backtrace for thread #0 (TID 35) ---
CrashDaemon(21:21): 0x3f5591ce: [libsystem.so] syscall2 +0xe (syscall.cpp:25 => syscall.cpp:24)
CrashDaemon(21:21): 0x8eff7134: [libc.so] raise +0x24 (syscall.h:34 => signal.cpp:20)
CrashDaemon(21:21): 0x8efe0f45: [libc.so] abort +0x29 (stdlib.cpp:230)
CrashDaemon(21:21): 0x8efe1a09: [libc.so] __assertion_failed +0xc9 (assert.cpp:31)
CrashDaemon(21:21): 0xa7f6c89d: [libweb.so] Web::Bindings::impl_from(JS::VM&, JS::GlobalObject&) +0x25d (WindowObject.cpp:107)
CrashDaemon(21:21): 0xa7f6f25d: [libweb.so] Web::Bindings::WindowObject::set_timeout(JS::VM&, JS::GlobalObject&) +0x3d (WindowObject.cpp:204)
CrashDaemon(21:21): 0x0fa5c12e: [libjs.so] AK::Function<JS::Value (JS::VM&, JS::GlobalObject&)>::CallableWrapper<JS::Value (*)(JS::VM&, JS::GlobalObject&)>::call(JS::VM&, JS::GlobalObject&) +0x2e (Function.h:151)
CrashDaemon(21:21): 0x0fadc2f9: [libjs.so] JS::NativeFunction::call() [clone .localalias] +0x79 (Function.h:91)
CrashDaemon(21:21): 0x0fbb2721: [libjs.so] JS::VM::call_internal(JS::FunctionObject&, JS::Value, AK::Optional<JS::MarkedValueList>) +0x671 (VM.cpp:655)
CrashDaemon(21:21): 0x0f9d78b3: [libjs.so] JS::CallExpression::execute(JS::Interpreter&, JS::GlobalObject&) const +0x373 (VM.h:301)
CrashDaemon(21:21): 0x0f9ce1fe: [libjs.so] JS::ExpressionStatement::execute(JS::Interpreter&, JS::GlobalObject&) const +0xae (AST.cpp:125)
CrashDaemon(21:21): 0x0fa114af: [libjs.so] JS::Interpreter::execute_statement(JS::GlobalObject&, JS::Statement const&, JS::ScopeType) +0xcf (Interpreter.cpp:180)
CrashDaemon(21:21): 0x0faf2e5d: [libjs.so] JS::OrdinaryFunctionObject::execute_function_body() [clone .localalias] +0x30d (OrdinaryFunctionObject.cpp:215)
CrashDaemon(21:21): 0x0faf3829: [libjs.so] JS::OrdinaryFunctionObject::call() +0x189 (OrdinaryFunctionObject.cpp:225)
CrashDaemon(21:21): 0x0fbb2721: [libjs.so] JS::VM::call_internal(JS::FunctionObject&, JS::Value, AK::Optional<JS::MarkedValueList>) +0x671 (VM.cpp:655)
CrashDaemon(21:21): 0x0f9d78b3: [libjs.so] JS::CallExpression::execute(JS::Interpreter&, JS::GlobalObject&) const +0x373 (VM.h:301)
CrashDaemon(21:21): 0x0f9ce1fe: [libjs.so] JS::ExpressionStatement::execute(JS::Interpreter&, JS::GlobalObject&) const +0xae (AST.cpp:125)
CrashDaemon(21:21): 0x0fa114af: [libjs.so] JS::Interpreter::execute_statement(JS::GlobalObject&, JS::Statement const&, JS::ScopeType) +0xcf (Interpreter.cpp:180)
CrashDaemon(21:21): 0x0f9cdf0b: [libjs.so] JS::Program::execute(JS::Interpreter&, JS::GlobalObject&) const +0xab (AST.cpp:93)
CrashDaemon(21:21): 0x0fa11a35: [libjs.so] JS::Interpreter::run(JS::GlobalObject&, JS::Program const&) +0x2e5 (Interpreter.cpp:60)
CrashDaemon(21:21): 0xa7fe845f: [libweb.so] Web::DOM::Document::run_javascript(AK::StringView const&, AK::StringView const&) +0xff (Document.cpp:681)
CrashDaemon(21:21): 0xa8051a1a: [libweb.so] Web::HTML::HTMLScriptElement::execute_script() +0x2aa (HTMLScriptElement.cpp:73 => HTMLScriptElement.cpp:41)
CrashDaemon(21:21): 0xa8052b54: [libweb.so] Web::HTML::HTMLScriptElement::prepare_script() [clone .localalias] +0x7e4 (HTMLScriptElement.cpp:292)
CrashDaemon(21:21): 0xa806f56c: [libweb.so] Web::HTML::HTMLDocumentParser::handle_text(Web::HTML::HTMLToken&) [clone .localalias] +0x17c (HTMLScriptElement.h:28 => HTMLDocumentParser.cpp:1893)
CrashDaemon(21:21): 0xa8080a4b: [libweb.so] Web::HTML::HTMLDocumentParser::run(AK::URL const&) [clone .localalias] +0x2cb (HTMLDocumentParser.cpp:139)
CrashDaemon(21:21): 0xa80e0682: [libweb.so] Web::FrameLoader::parse_document(Web::DOM::Document&, AK::Detail::ByteBuffer<32ul> const&) [clone .localalias] +0xb2 (FrameLoader.cpp:117)
CrashDaemon(21:21): 0xa80e3a41: [libweb.so] Web::FrameLoader::resource_did_load() +0xd41 (FrameLoader.cpp:264)
CrashDaemon(21:21): 0xa80eaf70: [libweb.so] Web::Resource::for_each_client(AK::Function<void (Web::ResourceClient&)>) [clone .localalias] +0x3b0 (Function.h:91)
CrashDaemon(21:21): 0xa80ec9a2: [libweb.so] Web::Resource::did_load(AK::Badge<Web::ResourceLoader>, AK::Span<unsigned char const>, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits, false> const&, AK::Optional<unsigned int>) +0x902 (Resource.cpp:102)
CrashDaemon(21:21): 0xa80ee94e: [libweb.so] AK::Function<void (AK::Span<unsigned char const>, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits, false> const&, AK::Optional<unsigned int>)>::CallableWrapper<Web::ResourceLoader::load_resource(Web::Resource::Type, Web::LoadRequest const&)::{lambda(auto:1, auto:2&, auto:3)#1}>::call(AK::Span<unsigned char const>, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits, false> const&, AK::Optional<unsigned int>) +0x6e (ResourceLoader.cpp:83)
CrashDaemon(21:21): 0xa80f1480: [libweb.so] AK::Function<void (bool, unsigned int, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits, false> const&, AK::Optional<unsigned int>, AK::Span<unsigned char const>)>::CallableWrapper<Web::ResourceLoader::load(Web::LoadRequest const&, AK::Function<void (AK::Span<unsigned char const>, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits, false> const&, AK::Optional<unsigned int>)>, AK::Function<void (AK::String const&, AK::Optional<unsigned int>)>)::{lambda(bool, auto:1, auto:2&, auto:3, AK::Span<unsigned char const>)#4}>::call(bool, unsigned int, AK::HashMap<AK::String, AK::String, AK::CaseInsensitiveStringTraits, false> const&, AK::Optional<unsigned int>, AK::Span<unsigned char const>) +0x370 (Function.h:91)
CrashDaemon(21:21): 0x1aa1a208: [libprotocol.so] AK::Function<void (bool, unsigned int)>::CallableWrapper<Protocol::Request::set_should_buffer_all_input(bool)::{lambda(auto:1, unsigned int)#2}>::call(bool, unsigned int) +0x288 (Function.h:91)
CrashDaemon(21:21): 0x1aa17cc1: [libprotocol.so] AK::Function<void ()>::CallableWrapper<Protocol::Request::stream_into(AK::OutputStream&)::{lambda()#2}>::call() +0x291 (Function.h:91)
CrashDaemon(21:21): 0x8358e789: [libcore.so] Core::Notifier::event(Core::Event&) [clone .localalias] +0x99 (Function.h:91)
CrashDaemon(21:21): 0x8358fd90: [libcore.so] Core::Object::dispatch_event(Core::Event&, Core::Object*) +0xb0 (Object.cpp:213)
CrashDaemon(21:21): 0x83577359: [libcore.so] Core::EventLoop::pump(Core::EventLoop::WaitMode) [clone .localalias] +0x3e9 (EventLoop.cpp:398)
CrashDaemon(21:21): 0x83577b6a: [libcore.so] Core::EventLoop::exec() +0x5a (EventLoop.cpp:362)
CrashDaemon(21:21): 0x11d64473: [/bin/WebContent] main +0x203 (main.cpp:43)
CrashDaemon(21:21): 0x11d645c7: [/bin/WebContent] _entry +0x57 (crt0.cpp:45)
```

### PR DESCRIPTION
**Pull request by davidot**
Most of the changes could have quite a big impact but everything seems to line up with the big browsers.

I left some comments to clarify some things

Fixes #8989 
And you can now go to http://css3test.com and run `window.onload()` in the js console to run the CSS3 tests
(It does crash fairly quickly in parse_calc_value)

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/StringPrototype.cpp [hunk 1]
```diff
@@ -1202,4 +1203,22 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::sup)
...
+    auto that_string = vm.argument(0).to_string(global_object);
+
+    // FIXME: Actually compare the string not just according to their bits.
+    if (string == that_string)
+        return Value(0);
+    if (string.value() < that_string)
+        return Value(-1);
+
+    return Value(1);
+}
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/StringPrototype.cpp:1222]: This is a super simple implementation but it actually passes all test262 tests.
So is this fine as a naive implementation?

File: Userland/Libraries/LibWeb/Bindings/WindowObject.cpp [hunk 2]
```diff
@@ -102,7 +102,18 @@ Origin WindowObject::origin() const
 
 static DOM::Window* impl_from(JS::VM& vm, JS::GlobalObject& global_object)
 {
-    auto* this_object = vm.this_value(global_object).to_object(global_object);
+    // Since this is a non built-in function we must treat it as non-strict mode
+    // this means that a nullish this_value should be converted to the
+    // global_object. Generally this does not matter as we try to convert the
+    // this_value to a specific object type in the bindings. But since window is
+    // the global object we make an exception here.
+    // This allows calls like `setTimeout(f, 10)` to work.
```
Comment (davidot) [inline Userland/Libraries/LibWeb/Bindings/WindowObject.cpp:110]: I'm not completely happy with this comment but did not want to go on and on.
If you know of a better way to phrase this or some other things I should mention please say so.

Also do we maybe want a comment in VM.cpp about naive functions being strict if they are part of the ECMAScript spec?

File: Userland/Libraries/LibWeb/CodeGenerators/WrapperGenerator.cpp [hunk 3]
```diff
@@ -1311,7 +1317,7 @@ namespace Web::Bindings {
 void @prototype_class@::initialize(JS::GlobalObject& global_object)
 {
     [[maybe_unused]] auto& vm = this->vm();
-    [[maybe_unused]] u8 default_attributes = JS::Attribute::Enumerable | JS::Attribute::Configurable;
+    [[maybe_unused]] u8 default_attributes = JS::Attribute::Enumerable | JS::Attribute::Configurable | JS::Attribute::Writable;
```
Comment (davidot) [inline Userland/Libraries/LibWeb/CodeGenerators/WrapperGenerator.cpp:1320]: This might seem dangerous but you already cannot overwrite readonly attributes since they have no setter and from my testing I could overwrite everything in the big browsers

File: Userland/Libraries/LibWeb/CodeGenerators/WrapperGenerator.cpp [hunk 4]
```diff
@@ -813,6 +813,12 @@ class @wrapper_class@ : public @wrapper_base_class@ {
 )~~~");
     }
 
+    if (interface.extended_attributes.contains("CustomHas")) {
```
Comment (linusg) [inline Userland/Libraries/LibWeb/CodeGenerators/WrapperGenerator.cpp:816]: Let's call this `HasProperty` for consistency.

File: Userland/Libraries/LibWeb/Bindings/WindowObject.cpp [hunk 5]
```diff
@@ -102,7 +102,18 @@ Origin WindowObject::origin() const
...
+    auto this_value = vm.this_value(global_object);
+    if (this_value.is_nullish()) {
+        this_value = global_object.value_of();
+    }
+
+    auto* this_object = this_value.to_object(global_object);
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Bindings/WindowObject.cpp:116]: There's no way this can fail anymore, so the below should really be `VERIFY(this_object)` instead.

File: Userland/Libraries/LibWeb/Bindings/WindowObject.cpp (hunk 2 above)
Comment (linusg) [reply Userland/Libraries/LibWeb/Bindings/WindowObject.cpp:110]: seems fine :)

File: Userland/Libraries/LibJS/Runtime/StringPrototype.cpp [hunk 6]
```diff
@@ -1202,4 +1203,22 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::sup)
...
 }
 
+// 22.1.3.10 String.prototype.localeCompare ( that [ , reserved1 [ , reserved2 ] ] ), https://tc39.es/ecma262/#sec-string.prototype.localecompare
+JS_DEFINE_NATIVE_FUNCTION(StringPrototype::locale_compare)
+{
+    auto string = ak_string_from(vm, global_object);
+    if (!string.has_value())
+        return {};
+
+    auto that_string = vm.argument(0).to_string(global_object);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/StringPrototype.cpp:1213]: Missing exception check

File: Userland/Libraries/LibJS/Runtime/StringPrototype.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/StringPrototype.cpp:1222]: yeah this is fine, but please add this comment below `// 22.1.3.10 ...`

```
// NOTE: This is the minimum localeCompare implementation for engines without ECMA-402.
```

so that we'll have an easier time finding all the functions that need updating when we implement ECMA-402.

File: Userland/Libraries/LibJS/Tests/builtins/String/String.prototype.localeCompare.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    expect(String.prototype.localeCompare).toHaveLength(1);
+
+    expect("".localeCompare("")).toBe(0);
+    expect("a".localeCompare("a")).toBe(0);
+    expect("6".localeCompare("6")).toBe(0);
+
+    expect("".localeCompare("a")).not.toBe(0);
+    expect("a".localeCompare("b")).not.toBe(0);
+    expect("6".localeCompare("7")).not.toBe(0);
+    expect("a".localeCompare("A")).not.toBe(0);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/builtins/String/String.prototype.localeCompare.js:11]: As mentioned before, please prefer testing exact outcomes and not just `not foo`.

File: Userland/Libraries/LibJS/Tests/builtins/String/String.prototype.localeCompare.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    expect("undefined".localeCompare()).toBe(0);
+    expect("undefined".localeCompare(undefined)).toBe(0);
+
+    expect("null".localeCompare(null)).toBe(0);
+    expect("null".localeCompare(undefined)).not.toBe(0);
+    expect("null".localeCompare()).not.toBe(0);
+
+    expect(() => {
+        String.prototype.localeCompare.call(undefined, undefined);
+    }).toThrow(TypeError);
```
Comment (linusg) [inline Userland/Libraries/LibJS/Tests/builtins/String/String.prototype.localeCompare.js:22]: Please include messages in error testing (`toThrowWithMessage`), unless there's a good reason not to (which isn't the case here)

File: Userland/Libraries/LibWeb/CodeGenerators/WrapperGenerator.cpp (hunk 3 above)
Comment (linusg) [reply Userland/Libraries/LibWeb/CodeGenerators/WrapperGenerator.cpp:1320]: Looking at the [spec](https://heycam.github.io/webidl/#es-interfaces) is much more useful than looking at other implementations :)

TL;DR: No, this is not correct in all cases, but neither was the previous code. Whatever, we can tighten this later.

**Review by linusg — CHANGES_REQUESTED** [review]

**Review by linusg — APPROVED** [review]
